code/modules/antagonists/heretic/knowledge/_heretic_paths.dm

```diff
@@ -0,0 +1,187 @@
+//Global typecache of all heretic knowledges -> instantiate the tree columns -> make them link themselves -> replace the old heretic stuff
+
+//heretic research tree is a directional graph so we can use some basic graph stuff to make internally handling it easier
+GLOBAL_LIST_INIT(heretic_research_tree,generate_heretic_research_tree())
```

Comment:
We can probably spare initializing this until someone gains a heretic antag datum, right?